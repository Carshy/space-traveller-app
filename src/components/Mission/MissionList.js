import React, { useEffect } from 'react';
import { useSelector, useDispatch, shallowEqual } from 'react-redux';
import { LoadMissions } from '../../redux/mission/mission';
import MissionPage from './Mission';
import './mission.css';

const Missions = () => {
  const missions = useSelector((state) => state.missionReducer, shallowEqual);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(LoadMissions());
  }, []);

  return (
    <section className="mission-section">
      <div className="header-container">
        <p className="mission">Mission</p>
        <p className="description">Description</p>
        <p className="status">Status</p>
        <p className="join">Join</p>
      </div>
      <div className="mission-container">
        { missions.map((mission) => (
          <MissionPage
            key={mission.mission_id}
            name={mission.mission_name}
            description={mission.description}
            id={mission.mission_id}
          />
        ))}
      </div>
    </section>
  );
};

export default Missions;
