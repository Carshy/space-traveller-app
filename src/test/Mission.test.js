import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/ConfigureStore';
import MissionList from '../components/Mission/MissionList';

test('renders correctly', () => {
  const tree = Renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <MissionList />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
