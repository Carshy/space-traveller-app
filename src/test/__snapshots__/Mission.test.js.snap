// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="mission-section"
>
  <div
    className="header-container"
  >
    <p
      className="mission"
    >
      Mission
    </p>
    <p
      className="description"
    >
      Description
    </p>
    <p
      className="status"
    >
      Status
    </p>
    <p
      className="join"
    >
      Join
    </p>
  </div>
  <ul
    className="mission-container>"
  />
</section>
`;
